### PR TITLE
Enhancement: Make footers translatable.

### PR DESCRIPTION
**Pull request by juanfra**
**Description**

I am adding i18n functions for footers.

I'm still doubtful we should include `Twenty Twenty-Five` as a hardcoded string in some footers. I know this discussion happened before. I believe that now that we have the copyright block binding, that's a good candidate. Else, re-consider the site title.

**Screenshots**

N/A

**Testing Instructions**

1. Open the site editor
2. Try every footer
3. Check that the strings are correct.

### CODE REVIEW

File: patterns/footer-centered.php [hunk 1]
```diff
@@ -23,7 +24,17 @@
...
-	<p class="has-text-align-center has-small-font-size">Designed with <a href="https://wordpress.org">WordPress</a></p>
+	<p class="has-text-align-center has-small-font-size">
+		<?php
+		/* Translators: WordPress link. */
+		$wordpress_link = '<a href="' . esc_url( __( 'https://wordpress.org', 'twentytwentyfive' ) ) . '" rel="nofollow">WordPress</a>';
+		printf(
+			/* Translators: Designed with WordPress */
+			esc_html__( 'Designed with %1$s', 'twentytwentyfive' ),
+			$wordpress_link // phpcs:ignore
+		);
```
Comment (carolinan) [inline patterns/footer-centered.php:32]: What is the purpose of using a variable when it is only used once?

Comment (juanfra) [reply patterns/footer-centered.php:32]: I'd say the purpose is readability. TT4 is using it like this. We can avoid the variable if you prefer it that way.

Comment (carolinan) [reply patterns/footer-centered.php:32]: By avoiding it, we could also avoid the `phpcs:ignore`.

Comment (juanfra) [reply patterns/footer-centered.php:32]: That's true, although there's nothing wrong with using `phpcs:ignore` per se. I'll apply the chances. I consider that this way is more readable, but that may be subjective.

Thanks.

**Comment by carolinan** [comment]
I agree that the theme name should not be in the footer. It is common in commercial themes, but in this case, I don't think WordPress needs to strengthen the "Twenty Twenty-Five" name, and the WordPress credit is already in the footer. But I'm not a marketer.

**Comment by carolinan** [comment]
This can be merged, if we want to handle the theme name text in a separate issue.

**Review by carolinan — APPROVED** [review]

**Comment by juanfra** [comment]
Thanks, @carolinan. I believe we can keep that discussion on an issue to hear different perspectives and merge this one.

**Pull request merged**